### PR TITLE
add shader deletion

### PR DESCRIPTION
**Pull request by munrocket**
works like a charm.

### CODE REVIEW

**Comment by davidar** [comment]
Profile pages don't list any shaders when you aren't logged in:

image

**Comment by munrocket** [comment]
This is privacy :D

**Comment by munrocket** [comment]
Probably not the best solution for hiding 'unlisted' shaders also 3 rerenderers.
But we need to check *user* and *id* for sure, maybe better to fix rerenderers for editor itself first.

**Comment by munrocket** [comment]
Maybe better to fix 'unlisted' in another PR. It's not so easy.

**Comment by munrocket** [comment]
@davidar do you have any other issues with this PR?

**Comment by davidar** [comment]
Sorry, haven't had a chance to review it properly yet, will try to do that soon. Could you fix the merge conflict?

File: components/profileshaders.tsx [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import { Button } from '@mui/material';
```
Comment (davidar) [inline components/profileshaders.tsx:1]: Please import Button directly like the other imports. When you do it like this it slows down builds quite a lot (which is stupid but we have to deal with it): https://mui.com/material-ui/guides/minimizing-bundle-size/

**Review by davidar — CHANGES_REQUESTED** [review]

File: components/profileshaders.tsx (hunk 1 above)
Comment (munrocket) [reply components/profileshaders.tsx]: Fixed.

**Review by davidar — APPROVED** [review]

**Pull request merged**